File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.py
def solution(dartResult):
    dartResult=list(dartResult)
    l=len(dartResult)
    i=0
    while i<l:
        if dartResult[i]=="S":
            dartResult[i]="**1+"
        elif dartResult[i]=="D":
            dartResult[i]="**2+"
        elif dartResult[i]=="T":
            dartResult[i]="**3+"
        elif dartResult[i]=="*":
            dartResult[i]="*2+"
            if i!=2:
                dartResult.insert(i-2,"*2+")
                l+=1
        elif dartResult[i]=="#":
            dartResult[i]="*(-1)+"
        i+=1
    return eval("".join(dartResult).replace("+*","*")[:-1])

File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.js
function solution(dartResult) {
    dartResult=dartResult.split("")
    let l=dartResult.length
    let i=0
    let dic={'S':'**1+','D':'**2+','T':'**3+','#':'*(-1)+'}
    while(i<=l){
        if(dartResult[i] in dic){
            dartResult[i]=dic[dartResult[i]]
        }
        else if(dartResult[i]==='*'){
            dartResult[i]="*2+"
            if(i!=2){
                dartResult.splice(i-2,0,"*2+")
            l+=1
            }
        }
        i+=1
    }
    let result=dartResult.join("").replaceAll("+*","*")
    return eval(result.slice(0,result.length-1))
}